### PR TITLE
feat: add code editor

### LINKED ISSUE #38: code editor component
### Clear and concise description of the problem

增加代码编写组件，语言包括但不限于javascript、sql、typescript、java...

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hongl-1**
### What is the purpose of this pull request? 

- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).

close #38

### CODE REVIEW

File: src/views/demo/code-editor/index.vue [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  }
+]
+
+function onChangeTheme(theme: string) {
+  message.success(`操作成功!当前主题为:${theme}`)
+}
+function onChangeLang(lang: string) {
+  message.success(`操作成功!当前主题为:${lang}`)
+}
+function onCodeChange(code: string) {}
```
Comment (Hongbusi) [inline src/views/demo/code-editor/index.vue:86]: 这里可以 `console.log` 一下，不然存在没什么意义了

File: src/views/demo/code-editor/index.vue [hunk 2]
```diff
@@ -0,0 +1,87 @@
+<template>
+  <page-wrapper title="code编辑器">
```
Comment (Hongbusi) [inline src/views/demo/code-editor/index.vue:2]: code 与编辑器之间可以添加一个空格

**Review by Hongbusi — CHANGES_REQUESTED** [review]

**Comment by Hongbusi** [comment]
image
需要删除中间的 issue，关联语法为：`close #id`

**Comment by Hongbusi** [comment]
image
顶部的标题可以加上 type。例如：`feat: add code editor`

**Comment by Hongbusi** [comment]
文档记得安排一下～

File: src/components/CodeEditor/src/CodeEditor.vue [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+</template>
+
+<script setup lang="ts">
+import 'codemirror/theme/ambiance.css'
+import 'codemirror/lib/codemirror.css'
+import 'codemirror/addon/hint/show-hint.css'
+import 'codemirror/addon/edit/matchbrackets'
+import 'codemirror/addon/selection/active-line'
+import 'codemirror/addon/hint/show-hint'
+import CodeMirror from 'codemirror/lib/codemirror'
```
Comment (likui628) [inline src/components/CodeEditor/src/CodeEditor.vue:17]: import CodeMirror from 'codemirror'

**Review by likui628 — COMMENTED** [review]
建议导入 @types/codemirror，避免 could not find a declaration file warning

File: src/components/CodeEditor/src/CodeEditor.vue [hunk 4]
```diff
@@ -0,0 +1,108 @@
+<template>
+  <div class="code-mirror" :style="{width: widthComputed, height: heightComputed}">
```
Comment (Hongbusi) [inline src/components/CodeEditor/src/CodeEditor.vue:2]: `:style="getStyle"` 可以把 widthComputed，heightComputed 合并一下

File: src/views/demo/code-editor/index.vue [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  }
+]
+
+function onChangeTheme(theme: string) {
+  message.success(`操作成功!当前主题为:${theme}`)
+}
+function onChangeLang(lang: string) {
+  message.success(`操作成功!当前主题为:${lang}`)
+}
+
```
Comment (Hongbusi) [inline src/views/demo/code-editor/index.vue:85]: 结尾多余空行可以删除

Comment (hongl-1) [reply src/views/demo/code-editor/index.vue:85]: OK 我改一下

**Review by Hongbusi — APPROVED** [review]

**Pull request merged**